RockLib.DistributedTracing.AspNetCore/AspNetCore/HttpContextExtensions.cs

```diff
@@ -46,16 +50,25 @@ public static ICorrelationIdAccessor GetCorrelationIdAccessor(this HttpContext h
             throw new ArgumentNullException(nameof(correlationIdHeader));
          }
 
-         if (!httpContext.Items.TryGetValue(typeof(ICorrelationIdAccessor), out var value)
-             || value is not ICorrelationIdAccessor accessor)
+         if (!httpContext.Items.TryGetValue(typeof(ICorrelationIdAccessor), out var value) 
+            || value is not ICorrelationIdAccessor accessor)
          {
             accessor = new CorrelationIdAccessor();
 
+            // align w/OpenTelemetry
+            accessor.TraceId = GetTraceId(httpContext);
+            accessor.SpanId = GetSpanId(httpContext);
+
             if (httpContext.GetHeaderValue(correlationIdHeader) is StringValues correlationId && correlationId.Count > 0)
             {
                accessor.CorrelationId = correlationId;
             }
 
+            if (accessor.CorrelationId is null)
```

Comment:
I don't think we should mix values in here for the same reasons as not marking CorrelationId obsolete.